### PR TITLE
Hybrid Search Tutorial

### PR DESCRIPTION
**Pull request by NirantK**

### CODE REVIEW

**Comment by Anush008** [comment]
Could we turn this to use the hybrid search interface introduced in https://github.com/qdrant/qdrant-client/pull/553?

**Comment by NirantK** [comment]
@Anush008 I'd prefer to do that in a separate notebook once a new Qdrant client release is made with these changes. If you'd like, please feel free to raise a PR for that as well?

**Comment by joein** [comment]
> @Anush008 I'd prefer to do that in a separate notebook once a new Qdrant client release is made with these changes. If you'd like, please feel free to raise a PR for that as well?

it is available as of qdrant-client == 1.8.2

AFAIK, @generall wanted to do some example with it as well, not sure what’s the status now

**Comment by NirantK** [comment]
> AFAIK, @generall wanted to do some example with it as well, not sure what’s the status now

I will add another notebook in a fresh PR with the latest Qdrant Client changes for hybrid and sparse support both. I need this for the docs as well as some questions from users

cc @joein @generall

File: docs/examples/Hybrid_Search.ipynb [hunk 1]
```diff
@@ -0,0 +1,1171 @@
...
+    "2. Preview data: Load and preview the data\n",
+    "3. Create Sparse Embeddings: Create SPLADE++ embeddings for the data\n",
+    "4. Create Dense Embeddings: Create BGE-Large-en-v1.5 embeddings for the data\n",
+    "5. Indexing: Index the embeddings using Qdrant\n",
+    "6. Search: Perform Hybrid Search using FastEmbed & Qdrant\n",
+    "7. Ranking: Rank the search results with Reciprocal Rank Fusion (RRF)\n",
+    "\n",
+    "## Setup\n",
+    "\n",
+    "In order to get started, you need only two dependencies, and we'll install them next:"
```
Comment (Anush008) [inline docs/examples/Hybrid_Search.ipynb:29]:
```suggestion
    "In order to get started, you need a few dependencies, and we'll install them next:"
```
The next step installs 4 direct deps.

File: docs/examples/Hybrid_Search.ipynb [hunk 2]
```diff
@@ -0,0 +1,1171 @@
...
+       "[]"
+      ]
+     },
+     "execution_count": 16,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "[x for x in product_texts if not isinstance(x, str)]"
```
Comment (Anush008) [inline docs/examples/Hybrid_Search.ipynb:644]: Is this left-over from testing?

File: docs/examples/Hybrid_Search.ipynb [hunk 3]
```diff
@@ -0,0 +1,1171 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "SparseVector(indices=[1, 2, 3], values=[0.1, 0.2, 0.3])"
```
Comment (Anush008) [inline docs/examples/Hybrid_Search.ipynb:790]: Left-over from testing?

File: docs/examples/Hybrid_Search.ipynb [hunk 4]
```diff
@@ -0,0 +1,1171 @@
...
+    "    points = []\n",
+    "    for idx, (text, sparse_vector, dense_vector) in enumerate(zip(product_texts, sparse_vectors, dense_vectors)):\n",
+    "        # print(sparse_vector)\n",
+    "        sparse_vector = SparseVector(indices=sparse_vector.indices.tolist(), values=sparse_vector.values.tolist())\n",
+    "        point = PointStruct(\n",
+    "            id=idx,\n",
+    "            payload={\"text\": text, \"product_id\": rows[idx][\"product_id\"]},  # Add any additional payload if necessary\n",
+    "            vector={\n",
+    "                \"text-sparse\": sparse_vector,\n",
+    "                \"text-dense\": dense_vector.tolist(),\n",
```
Comment (Anush008) [inline docs/examples/Hybrid_Search.ipynb:822]:
```suggestion
    "                \"text-dense\": dense_vector,\n",
```
np arrays can be passed I think.

**Pull request merged**